freqtrade/resolvers/hyperopt_resolver.py

```diff
@@ -20,17 +20,21 @@ class HyperOptResolver(IResolver):
     """
     This class contains all the logic to load custom hyperopt class
     """
-
     __slots__ = ['hyperopt']
 
-    def __init__(self, config: Dict) -> None:
+    def __init__(self, config: Dict = None) -> None:
```

Comment:
Is there a case where HyperoptResolver can be called without configuration (apart from tests)? 

I mean - we call `HyperOptResolver()` exactly once from non-test code, which always uses `self.config` as parameter - so having a default here seems to be misleading (especially since calling this class without config will result in an `OperationalException` 2 lines below this)

Reply:
I also noticed this, but left for a while -- it's here for more than an year...

Reply:
I copied it from the `HyperOptLossResolver.__init__` which is below. This should probably be removed everywhere in the resolvers...

Reply:
Done